js/main.js

```diff
@@ -1,97 +1,638 @@
+const form = document.querySelector('form');
+const summaryTab = document.querySelector('#summary-tab');
+const progressIndicator = document.querySelector('#progress-indicator');
+const contentBox = document.querySelector('.content-box');
+const error = document.querySelector('#error-messages');
 
 
+const getActiveTab = () => document.querySelector('section[id*="' + tabIdentifier + '"]:not(.hide)');
 
-        // console.log(item.id + ": index[" + currentTabNumber + "]");
-        // console.log("current tab: " + item.id + " - active tab: " + activeTab.id);
+const getTabs = () => document.querySelectorAll('section[id*="' + tabIdentifier + '"]');
 
-        // loop through all elements that match query selector and only continue for the current active tab
-        if (item.id === activeTab.id) {
-            if (type === 'next') {
-                arrayOfTabs[index + 1].classList.remove("hide");
-                var adjTabNumber = currentTabNumber + 1;
-            } else if (type === 'prev') {
-                arrayOfTabs[index - 1].classList.remove("hide");
-                var adjTabNumber = currentTabNumber - 1;
+const pushValuesToArray = (array, ref, elementName, elementAttribute, count) => {
+    array.push({
+        "ref": ref,
+        "prefix": elementName,
+        "attribute": elementAttribute,
+        "count": count,
+        "current": 1
+    });
+};
+
+const getAttributePrefix = attributeValue => {
+    if (typeof (attributeValue) == "string") {
+        let splitStringPosition = parseInt(attributeValue.lastIndexOf('-')) + 1;
+
+        return attributeValue.substr(0, splitStringPosition);
+    }
+};
+
+const initTab = (startTab = 0) => {
+    // function which will set the current tab
+    const allTabs = getTabs();
+    const numberOfTabs = allTabs.length;
+
+    let maxTab = numberOfTabs - 2; // array starts at zero therefore -1; another -1 as I want the final tab not to be accessible by next button, only by submit button.
+
+    if (startTab > maxTab) {
+        console.log(`ERROR: initTab function called for (tab: ${startTab}) which does not exist. Defaulting to tab(${maxTab}).`);
+        startTab = maxTab;
+    } else if (startTab < 0) {
+        console.log(`ERROR: initTab function called for (tab: ${startTab}) which does not exist. Defaulting to tab(0).`);
+        startTab = 0;
+    }
+
+    for (let i = 0; i < numberOfTabs; i++) {
+        if (i === startTab) {
+            allTabs[i].classList.remove('hide');
+            allTabs[i].classList.add('slide-in-left');
+        } else {
+            allTabs[i].classList.add('hide');
+        }
+    }
+
+    updateNavButtons();
+    updateTabIndicator();
+};
+
+const updateNavButtons = () => {
+    const numberOfTabs = getTabs().length - 1;
+    const activeTab = Array.from(getTabs()).indexOf(getActiveTab());
+
+    nextBtn.classList.add('hide');
+    prevBtn.classList.add('hide');
+    submitBtn.classList.add('hide');
```

Comment:
if you had ten buttons to hide, would you still do it this way? if not, I'ld like to see how you would do that here.